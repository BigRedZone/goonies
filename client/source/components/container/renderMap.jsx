import React from 'react';
import { GoogleApiWrapper, Map, Marker } from 'google-maps-react';

import key from '../../../../myapikey';

const apiKey = key.key;
// const apiKey = `${process.env.REACT_APP_GOOGLE_MAPS_KEY}`;

class GoogleMapsContainer extends React.Component {
  constructor(props) {
    super(props);
    const { google } = this.props;
    this.state = {
      activeMarker: {},
      selectedPlace: {},
    };
    // binding this to event-handler functions
    this.onMarkerClick = this.onMarkerClick.bind(this);
  }

  onMarkerClick() {
    this.setState({
      placeholder: 'placeholder',
      // selectedPlace: props,
      // activeMarker: marker,
    });
  }

  render() {
    const style = {
      width: '50vw',
      height: '75vh',
      marginLeft: 'auto',
      marginRight: 'auto',

    };
    return (
      <Map
        style={style}
        google={this.props.google}
        zoom={9}
        placeholder={this.state.placeholder}
        initialCenter={{ lat: 37.749669, lng: -119.555108 }}

      >
        <Marker
          onClick={this.onMarkerClick}
          title="Changing Colors Garage"
          position={{ lat: 37.749669, lng: -119.555108 }}
          name="Changing Colors Garage"
        />

      </Map>
    );
  }
}

export default GoogleApiWrapper({ apiKey })(GoogleMapsContainer);
