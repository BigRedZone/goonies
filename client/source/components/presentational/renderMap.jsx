import React from 'react';
import {
  GoogleApiWrapper, InfoWindow, Map, Marker,
} from 'google-maps-react';

import key from '../../../../myapikey.js';

const apiKey = key.key;
// const apiKey = `${process.env.REACT_APP_GOOGLE_MAPS_KEY}`;

class GoogleMapsContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      activeMarker: {},
      selectedPlace: {},
    };
    // binding this to event-handler functions
    this.onMarkerClick = this.onMarkerClick.bind(this);
  }

  onMarkerClick(props, marker, e) {
    this.setState({
      selectedPlace: props,
      activeMarker: marker,
    });
  }

  render() {
    const style = {
      width: '50vw',
      height: '75vh',
      marginLeft: 'auto',
      marginRight: 'auto',

    };
    return (
      <Map
        style={style}
        google={this.props.google}
        zoom={9}
        initialCenter={{ lat: 37.749669, lng: -119.555108 }}

      >
        <Marker
          onClick={this.onMarkerClick}
          title="Changing Colors Garage"
          position={{ lat: 37.749669, lng: -119.555108 }}
          name="Changing Colors Garage"
        />

      </Map>
    );
  }
}

export default GoogleApiWrapper({ apiKey })(GoogleMapsContainer);
