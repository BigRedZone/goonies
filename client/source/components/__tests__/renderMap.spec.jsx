import React from 'react';
import renderer from 'react-test-renderer';
import renderMap from '../presentational/renderMap';

test('RenderMap Renders correctly', () => {
  const component = renderer.create(<renderMap />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
