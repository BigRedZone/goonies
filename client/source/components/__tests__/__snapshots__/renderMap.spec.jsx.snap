// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderMap Renders correctly 1`] = `<renderMap />`;
